### DISCUSSION TITLE
Failed: static export + dynamic routes + a subpath (assetsPrefix)

### DISCUSSION THREAD

**Discussion by Izhaki** [vercel/next.js#10783] [Help] [answered]
Use case:

- A site with dynamic routes (catch-all case). 
- Deployed onto a subpath on the production server.
- Static export

# Non-production builds (works)

With a non-production build, this is the exported path map:

```
{
  '/grid.html': {
    page: '/grid'
  },
  '/dyn/what/ever.html': {
    page: '/dyn/[...slug]'
  }
}
```

# Production builds (fails)

However for production builds we add an assetsPrefix so the export path map ends like so:

```
{
  '/subpath/grid.html': {
    page: '/grid'
  },
  '/subpath/dyn/what/ever.html': {
    page: '/dyn/[...slug]'
  }
}
```
The first (regular) entry is fine, the second on (the dynamic one) fails with:

```
Error occurred prerendering page "/subpath/dyn/what/ever.html" 
https://err.sh/zeit/next.js/prerender-error: 
Error: The provided export path '/subpath/dyn/what/ever.html' doesn't match the '/dyn/[...slug]' page.
```

Any idea how this can be done?

**Comment by ijjk** [comment, marked as answer]
Hi, the `assetPrefix` isn't meant to be manually added to each path in the `exportPathMap` and is only meant for pointing the bundles/assets URLs to the correct location. If you want your pages nested you can add your export paths like normal (without the `assetPrefix` added to them) and then once export is complete deploy the contents of the `out` directory to the subdirectory you want it served under

**Comment by Izhaki** [comment]
I had to learn this the hard way.

I've created `subpath/dyn/[...slug]` for the export to work, only to then realise it does absolutely bugger all. The whole reason things were working was that all this was extending the defaultExportMap.

Thanks!
